hs/CalendarResponse.hs

```diff
@@ -0,0 +1,284 @@
+
+-- ** Event Creation
+
+-- | Generates an event for each course.
+getEvent :: String -> ([Time], String, String, String) -> [String]
+getEvent systemTime (timeFields, code, sect, session) =
+    concat $ eventsByCourse code sect session start end dates systemTime
+    where
+        dataInOrder = orderData timeFields
+        start = startTime dataInOrder
+        end = endTime dataInOrder
+        dates = getDatesByCourse session dataInOrder
+
+-- | Creates an event for each course depending on its session.
+eventsByCourse :: String -- ^ Course code.
+               -> String -- ^ Course section.
+               -> String -- ^ Course session.
+               -> [[String]] -- ^ Start times.
+               -> [[String]] -- ^ End times.
+               -> [[(String, String)]] -- ^ Start/End dates.
+               -> String -- ^ Current system time.
+               -> [[String]]
+eventsByCourse code sect "Y" start end dates systemTime =
+    eventsByTime code sect start end (dates !! 0) systemTime fallHolidays ++
+    eventsByTime code sect start end (dates !! 1) systemTime winterHolidays
+eventsByCourse code sect "F" start end dates systemTime =
+    eventsByTime code sect start end (dates !! 0) systemTime fallHolidays
+eventsByCourse code sect _ start end dates systemTime =
+    eventsByTime code sect start end (dates !! 0) systemTime winterHolidays
+
+-- | Creates an event for each start/end time and date.
+eventsByTime :: String -- ^ Course code.
+             -> String -- ^ Course section.
+             -> [[String]] -- ^ Start times.
+             -> [[String]] -- ^ End times.
+             -> [(String, String)] -- ^ Start/End dates.
+             -> String -- ^ Current system time.
+             -> [String] -- ^ Holidays dates.
+             -> [[String]]
+eventsByTime code sect start end dates systemTime holidays =
+    map (eventsByDate code sect systemTime holidays)
+        (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' (x:xs) (y:ys) (z:zs) = (x,y,z):zip' xs ys zs
+zip' _ _ _ = []
+
+-- | Creates an event for each start/end time.
+eventsByDate :: String -- ^ Course code.
+             -> String -- ^ Course section.
+             -> String -- ^ Current system time.
+             -> [String] -- ^ Holidays dates.
+             -> ([String], -- ^ Start time
+                 [String], -- ^ End time
+                 (String, String)) -- ^ Start/End date.
+             -> [String]
+eventsByDate code sect systemTime holidays (start, end, dates) =
+    concatMap (eventsStr code sect systemTime dates holidays) (zip start end)
+
+-- | Generates the string that represents each event.
+eventsStr :: String -- ^ Course code.
+          -> String -- ^ Course section.
+          -> String -- ^ Current system time.
+          -> (String, String) -- ^ Start/End date.
+          -> [String] -- ^ Holidays dates.
+          -> (String, String) -- ^ Start/End time.
```

Comment:
> In my opinion, it does not hurt to have more information about the function. However, if we use type synonyms, we should avoid writing repetitive information in the comments.

Correct. Redundancy is usually bad.

> I do not think that we need Type Synonyms here because I cannot think of any more meaningful information that we could add in the comments.

A principle can be derived here: if we have the opportunity to say something in the comments or through more expressive code, which one is better?

Also, there is now a type called `Code`, which is a synonym for a `String`. This function uses a `String` to represent a code. This may be a good opportunity for a change.